packages/extension-koni-ui/src/contexts/AuthenticationMythProvider.tsx

```diff
@@ -0,0 +1,171 @@
+
+export const AuthenticationMythProvider = ({ children }: AuthenticationMythProviderProps) => {
+  const [account, setAccount] = useState<AccountPublicInfo>({} as AccountPublicInfo);
+  const [validatedData, setValidateData] = useState<boolean>();
+  const [tokenData, setTokenData] = useState<TTokenData>();
+  const [linkData, setLinkData] = useState<LinkResult>();
+  const [isLinked, setIsLinked] = useState<boolean>(false);
+  const authContext = useContext(AuthContext);
+  const { currentAccount } = useSelector((state: RootState) => state.accountState);
+
+  const onLoginWithMythAccount = useCallback(() => {
+    authContext.logIn();
+  }, [authContext]);
+
+  const onLoginWithTelegramAccount = useCallback(async (address: string) => {
+    await bookaSDK.login(address);
+  }, []);
+
+  const onLogoutMythAccount = useCallback(() => {
+    authContext.logOut();
+  }, [authContext]);
+
+  const onSubmitMythAccount = useCallback(async (address: string) => {
+    if (!validatedData) {
+      throw new Error('Please validate data first');
+    }
+
+    if (!tokenData?.email || !authContext.token) {
+      Telegram.WebApp.showAlert('Please login first');
+
+      return;
+    }
+
+    const rs = await linkSDK.submitLink({
+      initData,
+      linkInfo: {
+        email: tokenData?.email as string,
+        token: authContext.token,
+        address
+      }
+    });
+
+    if (rs.error) {
+      Telegram.WebApp.showAlert(rs.error);
+    } else if (rs.success) {
+      setIsLinked(rs.success);
+      setLinkData(rs.data);
+    }
+  }, [authContext.token, tokenData?.email, validatedData]);
+
+  const linkMythAccount = useCallback(async (address: string) => {
+    if (!tokenData?.email || !authContext.token) {
+      onLoginWithMythAccount();
+    }
+
+    await onSubmitMythAccount(address);
+  }, [authContext.token, onLoginWithMythAccount, onSubmitMythAccount, tokenData?.email]);
+
+  const onLogin = useCallback(async (address: string, isWithoutMyth = false) => {
+    if (!isWithoutMyth) {
+      onLoginWithMythAccount();
+      await linkMythAccount(address);
+    } else {
+      await onLoginWithTelegramAccount(address);
+    }
+  }, [linkMythAccount, onLoginWithMythAccount, onLoginWithTelegramAccount]);
+
+  const onLogout = useCallback(async () => {
+    onLogoutMythAccount();
+
+    return Promise.resolve();
+  }, [onLogoutMythAccount]);
+
+  useEffect(() => {
+    setTokenData(authContext.tokenData);
+  }, [authContext.tokenData]);
+
+  useEffect(() => {
+    if (linkData) {
+      setAccount((prev) => {
+        return {
+          ...prev,
+          address: linkData.link_address,
+          email: linkData.link_email,
+          id: linkData.telegram_id,
+          signature: linkData.link_signature,
+          uid: linkData.link_uid
+        };
+      });
+    }
+  }, [linkData]);
+
+  useEffect(() => {
+    linkSDK.validateData({ initData }).then((rs) => {
+      if (rs.success) {
+        setValidateData(rs.data?.validData);
```

Comment:
This should be remove in real production. And update with default init data from env for development. Not require validate